scylla.go

```diff
@@ -413,15 +413,14 @@ func (p *scyllaConnPicker) maybeReplaceWithLessBusyConnection(c *Conn) *Conn {
 		return c
 	}
 	alternative := p.leastBusyConn()
-	if alternative == nil || alternative.AvailableStreams()*120 > c.AvailableStreams()*100 {
-		return c
-	} else {
+	if alternative != nil && alternative.StreamsInUse()*100 >= c.StreamsInUse()*(100-c.session.cfg.HeavyLoadedSwitchConnectionPercentage) {
```

Comment:
Shouldn't it be `<`?
Right now you are returning the alternative if it's more loaded than the original connection, not less

Reply:
It would be good to have some test to verify that the inequality is correct.

Reply:
both done, added tests and fixed the code